pkg/controller/machine/machine_controller.go

```diff
@@ -170,6 +178,15 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 		}
 
 		klog.Infof("Reconciling machine %q triggers delete", name)
+
+		// Drain node before deletion
+		// If a machine is not linked to a node, just delete the machine.
+		if _, exists := m.ObjectMeta.Annotations[clusterv1.ExcludeNodeDrainingAnnotation]; !exists && m.Status.NodeRef != nil {
```

Comment:
Can't combine in this branch with ```m.Stat.NodeRef != nil``` below because we delete instance before deleting node.